sql/core/src/test/scala/org/apache/spark/sql/StringFunctionsSuite.scala

```diff
@@ -657,4 +657,30 @@ class StringFunctionsSuite extends QueryTest with SharedSparkSession {
     }.getMessage
     assert(m.contains("data type mismatch: argument 1 requires string type"))
   }
+
+  test("SPARK-38063: string split_part function") {
```

Comment:
Can we also add some test cases that throw errors?

Reply:
done. There is only one case that throws exception/errors, which is index as 0 (thus throws ArrayIndexOutOfBoundsException).

If any input is null, or index is out of range (except 0), it returns null.

Reply:
+1

Reply:
What about index (pos or neg) larger than the number of split parts?

Reply:
added a few more test cases about both positive and negative partNum/index, and also cover a few more edge cases.